### ISSUE TITLE
Player already connected to proxy

### ISSUE THREAD

**Issue by DebitCardz** [VelocityPowered/Velocity#312]
I found an issue in Velocity 1.1.0 where if you have a command which uses `p.createConnectionRequest(<server>).connect();` and you also repeatedly use /server <server> after a while of spamming both commands you will be timed out of the server but the proxy will still think your connected to it which leads to you not being able to reconnect to it if you don't automatically kick the player if a player with the same name tries to connect.

Another side effect is that the Proxy will not properly shutdown and will take over 10 seconds to do so in this glitched state.

**Comment by astei** [comment]
Does fa954ab717 fix the issue (latest build)?

**Comment by DebitCardz** [comment]
Give me a moment to check.

**Comment by astei** [comment]
Looks like it's not fixed? I can't reproduce very easily on localhost, probably because the connection gets established nearly instantly. I will test on a geographically distant VPS.

**Comment by astei** [comment]
Most recent commit should have fixed this. I verified by running a torture test where a client would attempt to make connections from and to a server repeatedly for 1 minute. The client did not get timed out and `/glist` was normal as expected.

**Comment by DebitCardz** [comment]
Let me test on my side.

**Comment by DebitCardz** [comment]
It's still broken for me, after about 10 seconds of spamming /server test-1 then using my /hub command (which as stated before I use `p.createConnectionRequest(<server>).connect();` to create the connection) it timed me out and still had the invalid player in /glist.

**Comment by astei** [comment]
I am still unable to reproduce this problem. I spammed server connections for several seconds (all of which got rejected afterwards) and then tried teleporting back to the lobby of my local test server, which was successful. I even induced lag to the backend server to test, and that worked fine as well.

Without some more details about your setup I can't possibly hope to fix this issue.

**Comment by astei** [comment]
Closing the issue, this turned out to be another issue entirely (the Netty I/O threads were getting blocked by the command, triggering the issue).
